pkg/kp/rcstore/consul_store.go

```diff
@@ -158,17 +158,26 @@ func (s *consulStore) WatchNew(quit <-chan struct{}) (<-chan []fields.RC, <-chan
 		defer close(outCh)
 		defer close(errCh)
 
-		for listed := range inCh {
+		// Initialize one value off the inCh
+		listed := <-inCh
+
+		for {
 			out, err := s.kvpsToRCs(listed)
 			if err != nil {
 				select {
 				case errCh <- err:
 				case <-quit:
+					break
 				}
 			} else {
 				select {
```

Comment:
To avoid rewriting the same result on `outCh` repeatedly, perhaps check for an update with a second select right above this?
```
select {
   case listed = <- inCh:
   default:
}
``` 
?